charms/istio-pilot/config.yaml

```diff
@@ -24,3 +24,6 @@ options:
     type: string
     default: istio-ingressgateway-workload
     description: Name of the service created by istio-gateway to use as a Gateway
+  tls-secret-id:
+    type: secret
+    description: A configuration option to store the user secret ID that stores the TLS certificate and key values.
```

Comment:
Can we add to this description what the secret should contain?  The code looks for content in the `tls-crt` and `tls-key` keys, so we should explain that here.

Reply:
This is an awkward one because this configuration option is just for the secret id, not for the secret itself. Since we don't have a `secret.yaml` file or something similar, I think adding that explanation here could be misleading. Do you think a better place to put that info is in the public docs?

Reply:
For that, I think that we **could** add this information here. Since there is no other way to let users know what the contents of the secret should be. Apart from it, I think we should document it also in the charm's [README](https://github.com/canonical/istio-operators/tree/main/charms/istio-pilot).

Reply:
Would it be helpful if I add the secret description in the README and mention the secret contents should passed as mentioned in that doc?

Reply:
Sure this makes sense!

Reply:
Thinking about this more, since the docs are not published yet, I will just add a note in the description of the config option. I will then submit a PR for adding instructions in the README and in public docs. let me know what you think.